### PR TITLE
Refactoring explainer and plotting

### PR DESCRIPTION
**Pull request by ErikHartman**
Making explainer more robust and following better coding practices.

### CODE REVIEW

**Comment by ErikHartman** [comment]
- Removed importance_network
- Added normalization to explainer:
```py
from binn import BINNExplainer

explainer = BINNExplainer(binn)
single_explanations = explainer.explain_single(dataloaders, split="val")
normalized_single_explanations = explainer.normalize_importances(single_explanations, method="fan")
normalized_single_explanations
```

I haven't reworked the plotting functionality (Sankey) yet...

**Comment by ErikHartman** [comment]
- Moved sankey into `binn/plot`

**Comment by ErikHartman** [comment]
- Sankey is behaving nicely now.

File: binn/model/util.py [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-def load_reactome_db():
+def load_reactome_db(input_source = "uniprot"):
     """
     Load default Reactome mapping and pathways files.
     """
     base_dir = os.path.dirname(os.path.abspath(__file__))
 
-    mapping_path = os.path.join(base_dir, "../../data/uniprot_2_reactome_2025_01_14.txt")
-    pathways_path = os.path.join(base_dir, "../../data/reactome_pathways_relation_2025_01_14.txt")
+    if input_source == "uniprot":
```
Comment (arnscott) [inline binn/model/util.py:10]: I don't think this will work when the code is packaged and built as a whl. It's needs to be included in the MANIFEST or in setup.py as package_data, or imported in some other way, like this:

https://stackoverflow.com/questions/6028000/how-to-read-a-static-file-from-inside-a-python-package

File: binn/plot/util.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    """
+    Load a default DataFrame based on a known string key (e.g. 'reactome').
+    You can add more options if you have multiple defaults.
+
+    Example usage:
+        df = load_default_mapping('reactome')
+    """
+    base_dir = os.path.dirname(os.path.abspath(__file__))
+    if mapping_name.lower() == "reactome":
+        mapping_path = os.path.join(base_dir, "../../data/downloads/reactome_pathways_names_2025_01_21.txt")
```
Comment (arnscott) [inline binn/plot/util.py:14]: Same as the comment above

**Review by arnscott — CHANGES_REQUESTED** [review]

File: binn/analysis/explainer.py [hunk 3]
```diff
@@ -1,260 +1,378 @@
...
-            test_data (torch.Tensor): The input data for which to generate the explanations.
-            background_data (torch.Tensor): The background data to use for computing the Shapley values.
-            layer (int): The index of the layer for which to compute the SHAP explanations.
+        for name, layer in self.model.layers.named_children():
+            if isinstance(layer, nn.Linear):
+                explainer = shap.DeepExplainer((self.model, layer), background_data)
+                svals = explainer.shap_values(test_data)
+                shap_results["features"].append(self.model.layer_names[layer_idx])
+                shap_results["shap_values"].append(svals)
+                layer_idx += 1
```
[the remark this answers is not in the document]
Comment (ErikHartman) [reply binn/analysis/explainer.py:233]: Doesn't work with isinstance since enumerate will always increment even if isinstance is false

**Comment by ErikHartman** [comment]
Fixed comments @arnscott

**Review by arnscott — APPROVED** [review]
